client/src/main/java/org/eclipse/hono/client/Command.java

```diff
@@ -349,14 +355,45 @@ public static String getRequestId(final String correlationId, final String reply
             return null;
         }
 
-        final String stringOne = Optional.ofNullable(correlationId).orElse("");
-        String stringTwo = Optional.ofNullable(replyToId).orElse("");
-        final boolean removeDeviceFromReplyTo = stringTwo.startsWith(deviceId + "/");
-        if (removeDeviceFromReplyTo) {
-            stringTwo = stringTwo.substring(deviceId.length() + 1);
+        final String correlationIdOrEmpty = Optional.ofNullable(correlationId).orElse("");
+        String replyToIdWithoutDeviceOrEmpty = Optional.ofNullable(replyToId).orElse("");
+        final boolean replyToContainedDeviceId = replyToIdWithoutDeviceOrEmpty.startsWith(deviceId + "/");
+        if (replyToContainedDeviceId) {
+            replyToIdWithoutDeviceOrEmpty = replyToIdWithoutDeviceOrEmpty.substring(deviceId.length() + 1);
+        }
+        return String.format("%s%02x%s%s", encodeReplyToOptions(replyToContainedDeviceId), correlationIdOrEmpty.length(),
+                correlationIdOrEmpty, replyToIdWithoutDeviceOrEmpty);
+    }
+
+    /**
+     * Encodes the given boolean parameters related to the original reply-to address of the command message as a single
+     * digit string.
+     * 
+     * @param replyToContainedDeviceId Whether the original reply-to address of the command message contained the device
+     *            id.
+     * @return The encoded options as a single digit string.
+     */
+    static String encodeReplyToOptions(final boolean replyToContainedDeviceId) {
+        int bitFlag = 0;
+        if (replyToContainedDeviceId) {
+            bitFlag |= FLAG_REPLY_TO_CONTAINED_DEVICE_ID;
         }
-        return String.format("%s%02x%s%s", removeDeviceFromReplyTo ? "1" : "0", stringOne.length(), stringOne,
-                stringTwo);
+        return String.valueOf(bitFlag);
+    }
+
+    /**
+     * Returns {@code true} if the original reply-to address of the command message contained the device id.
```

Comment:
Can we agree to not describe the return value in the description of the method's purpose and do that in the `@return` tag instead? The description would then be something like
> Checks if the original reply-to address of the command message contained the device id.

Reply:
Ok, I've changed that accordingly.